src/base/types/derivedmodels/GeckoModel.jl

```diff
@@ -0,0 +1,423 @@
+            rid,
+            original_rid,
+            E_components,
+            col_idx,
+            gene_ids,
+        )
+    end
+
+    Se = sparse(
+        E_components.row_idxs,
+        E_components.col_idxs,
+        E_components.coeffs,
+        num_genes,
+        num_reactions,
+    )
+
+    stoich_mat = sparse([
+        S zeros(num_metabolites, num_genes)
+        Se I(num_genes)
+    ])
+
+    #: equality rhs
+    b = spzeros(num_metabolites + num_genes)
+
+    #: find objective (assume objective is forward)
+    obj_idx_orig = first(findnz(objective(model))[1])
+    obj_id_orig = reactions(model)[obj_idx_orig]
+    obj_id = obj_id_orig * "§FOR"
+    c = spzeros(num_vars)
+    obj_idx = reaction_map[obj_id]
+    c[obj_idx] = 1.0
+
+    #: inequality constraints
+    xl = sparse([lb_fluxes; fill(0.0, num_genes)])
+    xu = sparse([ub_fluxes; fill(1000.0, num_genes)])
+
+    #: enzyme capacity constraints
+    mw_proteins = [model.genes[pid].molar_mass for pid in gene_ids]
+    C = spzeros(length(enzyme_capacities), num_vars)
+    cl = spzeros(length(enzyme_capacities))
+    cu = spzeros(length(enzyme_capacities))
+
+    for (i, enz_cap) in enumerate(enzyme_capacities)
+        enz_idxs = indexin(first(enz_cap), gene_ids)
+        C[i, num_reactions.+enz_idxs] .= mw_proteins[enz_idxs]
+        cu[i] = last(enz_cap)
+    end
+
+    return GeckoModel(
+        reactions(model),
+        _order_id_to_idx_dict(reaction_map),
+        _order_id_to_idx_dict(metabolite_map),
+        gene_ids,
+        c,
+        stoich_mat,
+        b,
+        xl,
+        xu,
+        C,
+        cl,
+        cu,
+    )
+end
+
+"""
+    _add_enzyme_variable(
+        rid_isozymes,
+        iso_num,
+        rid,
+        original_rid,
+        E_components,
+        col_idx,
+        gene_ids,
+    )
+
+Helper function to add an column into the enzyme stoichiometric matrix.
+"""
+function _add_enzyme_variable(
+    rid_isozymes,
+    iso_num,
+    rid,
+    original_rid,
+    E_components,
+    col_idx,
+    gene_ids,
+)
+    pstoich = rid_isozymes[original_rid][iso_num].stoichiometry
+    kcat =
+        contains(rid, "§FOR") ? rid_isozymes[original_rid][iso_num].kcats[1] :
+        rid_isozymes[original_rid][iso_num].kcats[2]
+    for (pid, pst) in pstoich
+        push!(E_components.row_idxs, first(indexin([pid], gene_ids)))
+        push!(E_components.col_idxs, col_idx)
+        push!(E_components.coeffs, -pst / kcat)
+    end
+end
+
+"""
+    change_bound!(model::GeckoModel, id; lower=nothing, upper=nothing)
+
+Change the bound of variable in `model`. Does not change the bound if respective
+bound is `nothing`. Note, for `GeckoModel`s, if the model used to construct the
+`GeckoModel` has irreversible reactions, then these reactions will be
+permanently irreversible in the model, i.e. changing their bounds to make them
+reversible will have no effect.
+"""
+function change_bound!(model::GeckoModel, id; lower = nothing, upper = nothing)
```

Comment:
this should go to other reconstruction functions into `reconstructions/gecko.jl` or so

Reply:
(also the other `change_bounds!`)